fe/src/main/java/org/apache/doris/common/util/DynamicPartitionUtil.java

```diff
@@ -0,0 +1,255 @@
+            analyzedProperties.put(DynamicPartitionProperty.PREFIX, prefixValue);
+        }
+        if (properties.containsKey(DynamicPartitionProperty.END)) {
+            String endValue = properties.get(DynamicPartitionProperty.END);
+            checkEnd(endValue);
+            properties.remove(DynamicPartitionProperty.END);
+            analyzedProperties.put(DynamicPartitionProperty.END, endValue);
+        }
+        if (properties.containsKey(DynamicPartitionProperty.BUCKETS)) {
+            String bucketsValue = properties.get(DynamicPartitionProperty.BUCKETS);
+            checkBuckets(bucketsValue);
+            properties.remove(DynamicPartitionProperty.BUCKETS);
+            analyzedProperties.put(DynamicPartitionProperty.BUCKETS, bucketsValue);
+        }
+        if (properties.containsKey(DynamicPartitionProperty.ENABLE)) {
+            String enableValue = properties.get(DynamicPartitionProperty.ENABLE);
+            checkEnable(db, olapTable, enableValue);
+            properties.remove(DynamicPartitionProperty.ENABLE);
+            analyzedProperties.put(DynamicPartitionProperty.ENABLE, enableValue);
+        }
+        return analyzedProperties;
+    }
+
+    public static void checkAlterAllowed(OlapTable olapTable) throws DdlException {
+        TableProperty tableProperty = olapTable.getTableProperty();
+        if (tableProperty != null &&
+                tableProperty.getDynamicPartitionProperty().isExist() &&
+                Boolean.parseBoolean(tableProperty.getDynamicPartitionProperty().getEnable())) {
+            throw new DdlException("Cannot modify partition on a Dynamic Partition Table, set `dynamic_partition.enable` to false firstly.");
+        }
+    }
+
+    public static boolean isDynamicPartitionTable(Table table) {
+        if (!(table instanceof OlapTable) ||
+                !(((OlapTable) table).getPartitionInfo().getType().equals(PartitionType.RANGE))) {
+            return false;
+        }
+        RangePartitionInfo rangePartitionInfo = (RangePartitionInfo) ((OlapTable) table).getPartitionInfo();
+        TableProperty tableProperty = ((OlapTable) table).getTableProperty();
+        if (tableProperty == null || !tableProperty.getDynamicPartitionProperty().isExist()) {
+            return false;
+        }
+
+        String enable = tableProperty.getDynamicPartitionProperty().getEnable();
+        return rangePartitionInfo.getPartitionColumns().size() == 1 &&
+                !Strings.isNullOrEmpty(enable) && enable.equalsIgnoreCase(Boolean.TRUE.toString());
+    }
+
+    public static String getPartitionFormat(Column column) {
+        if (column.getDataType().equals(PrimitiveType.DATE)) {
+            return DATE_FORMAT;
+        } else if (column.getDataType().equals(PrimitiveType.DATETIME)) {
+            return DATETIME_FORMAT;
+        } else {
+            // TODO: How to resolve int type a better way
+            return TIMESTAMP_FORMAT;
```

Comment:
For int type, I use yyyyMMdd to format, for example, It will return` values less than "20191219"`, if user use timestamp they will find error msg in `show dynamic partition tables`